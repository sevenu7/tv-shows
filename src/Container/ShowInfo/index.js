import React, { Component } from "react";
import { Col, Container, Row, Spinner } from "react-bootstrap";
import InnerHtml from "dangerously-set-html-content";

import InfoCard from "../../Components/InfoCard";
// import axios from "axios";
import { fetchShow } from "../../services";

class ShowInfo extends Component {
  state = {
    show: this.props.location.show,
  };

  componentDidMount() {
    const { show } = this.state;
    console.log(this.props.location);
    const id = this.props.match.params.id;
    if (!show) {
      fetchShow(id)
        .then((res) => this.setState({ show: res.data }))
        .catch((err) => console.log(err));
    }
  }

  render() {
    const { show } = this.state;
    console.log(show);
    return show ? (
      <Container>
        <Row>
          <Col sm={12} md={8} lg={8} xl={8}>
            <h2>{show.name}</h2>
            <img
              className="infoImage"
              src={show.image.medium}
              alt={show.name}
            />
            <InnerHtml html={show.summary} />
          </Col>
          <Col sm={12} md={4} lg={4} xl={4}>
            <InfoCard show={show} />
          </Col>
        </Row>
      </Container>
    ) : (
      <Spinner className="spinner" animation="border" role="status">
        <span className="sr-only">Loading...</span>
      </Spinner>
    );
  }
}

export default ShowInfo;
