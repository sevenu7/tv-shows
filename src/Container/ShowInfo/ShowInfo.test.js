import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import axios from "axios";
import ShowInfo from "./";

jest.mock("axios");

configure({ adapter: new Adapter() });

let wrapper;
const location = {
  show: {
    id: 1,
    url: "http://www.tvmaze.com/shows/1/under-the-dome",
    name: "Under the Dome",
    type: "Scripted",
    language: "English",
    genres: ["Drama", "Science-Fiction", "Thriller"],
    status: "Ended",
    runtime: 60,
    premiered: "2013-06-24",
    officialSite: "http://www.cbs.com/shows/under-the-dome/",
    schedule: { time: "22:00", days: ["Thursday"] },
    rating: { average: 6.5 },
    weight: 97,
    network: {
      id: 2,
      name: "CBS",
      country: {
        name: "United States",
        code: "US",
        timezone: "America/New_York",
      },
    },
    webChannel: null,
    externals: { tvrage: 25988, thetvdb: 264492, imdb: "tt1553656" },
    image: {
      medium:
        "http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
      original:
        "http://static.tvmaze.com/uploads/images/original_untouched/81/202627.jpg",
    },
    summary:
      "<p><b>Under the Dome</b> is the story of a small town that is suddenly and inexplicably sealed off from the rest of the world by an enormous transparent dome. The town's inhabitants must deal with surviving the post-apocalyptic conditions while searching for answers about the dome, where it came from and if and when it will go away.</p>",
  },
};

const match = { params: { id: 1 } };
const data = { data: "test" };
axios.get.mockResolvedValue(data);

beforeEach(() => {
  wrapper = shallow(<ShowInfo location={{}} match={match} />);
});

describe("<ShowInfo />", () => {
  it("should match Snapshot without location prop and success response", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should match Snapshot without location prop and error response", () => {
    axios.get.mockRejectedValue(data);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should match Snapshot", () => {
    wrapper = shallow(<ShowInfo location={location} match={match} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
