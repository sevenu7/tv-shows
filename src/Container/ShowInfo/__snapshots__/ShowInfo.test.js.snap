// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowInfo /> should match Snapshot 1`] = `
"<Container fluid={false}>
  <Row noGutters={false}>
    <Col sm={12} md={8} lg={8} xl={8}>
      <h2>
        Under the Dome
      </h2>
      <img className=\\"infoImage\\" src=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" alt=\\"Under the Dome\\" />
      <DangerouslySetHtmlContent html=\\"<p><b>Under the Dome</b> is the story of a small town that is suddenly and inexplicably sealed off from the rest of the world by an enormous transparent dome. The town\\\\'s inhabitants must deal with surviving the post-apocalyptic conditions while searching for answers about the dome, where it came from and if and when it will go away.</p>\\" />
    </Col>
    <Col sm={12} md={4} lg={4} xl={4}>
      <InfoCard show={{...}} />
    </Col>
  </Row>
</Container>"
`;

exports[`<ShowInfo /> should match Snapshot without location prop and error response 1`] = `
"<Container fluid={false}>
  <Row noGutters={false}>
    <Col sm={12} md={8} lg={8} xl={8}>
      <h2 />
      <img className=\\"infoImage\\" src=\\"no-img-portrait-text.png\\" alt={[undefined]} />
      <DangerouslySetHtmlContent html={[undefined]} />
    </Col>
    <Col sm={12} md={4} lg={4} xl={4}>
      <InfoCard show=\\"test\\" />
    </Col>
  </Row>
</Container>"
`;

exports[`<ShowInfo /> should match Snapshot without location prop and success response 1`] = `
"<Container fluid={false}>
  <Row noGutters={false}>
    <Col sm={12} md={8} lg={8} xl={8}>
      <h2 />
      <img className=\\"infoImage\\" src=\\"no-img-portrait-text.png\\" alt={[undefined]} />
      <DangerouslySetHtmlContent html={[undefined]} />
    </Col>
    <Col sm={12} md={4} lg={4} xl={4}>
      <InfoCard show=\\"test\\" />
    </Col>
  </Row>
</Container>"
`;
